File: problems/java/integer-to-roman/_12_Integer_to_Roman.java
package solution._12_;

/**
 * Author: Sorcerer
 * Date: 2018/4/26
 * Description:
 */
public class _12_Integer_to_Roman {

    public String intToRoman(int num) {
        int[] nums = new int[]{1, 5, 10, 50, 100, 500, 1000,};
        char[] romans = new char[]{'I', 'V', 'X', 'L', 'C', 'D', 'M',};
        for (int i = nums.length - 1; i >= 0; i++) {

        }
        return "";
    }

    public static void main(String[] args) {
        _12_Integer_to_Roman solution = new _12_Integer_to_Roman();
        System.out.println(solution.intToRoman(1994));
    }
}

File: problems/kotlin/integer-to-roman/_12_Integer_to_Roman.kt
import java.util.*

/**
 * @author: Sorcerer
 * @date: 11/1/2018
 * @description:
 */
class _12_Integer_to_Roman {
    private val romans = arrayOf(
            "M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"
    )
    private val nums = arrayOf(
            1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1
    )

    fun intToRoman(num: Int): String {
        var rest = num
        val builder = StringBuilder()
        for (i in 0 until romans.size) {
            if (nums[i] <= rest) {
                val cnt = rest / nums[i]
                rest -= cnt * nums[i]
                for (j in 0 until cnt) builder.append(romans[i])
            }
        }
        return builder.toString()
    }
}

fun main(args: Array<String>) {
    var question = 1
    while (question <= 3999) {
        println("$question: " + _12_Integer_to_Roman().intToRoman(question))
        question += Random().nextInt(20)
    }
}
